src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,129 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_START_CAPACITY = 16;
+    private static final int INCREASE_CAPACITY_VALUE = 2;
+    private int capacity;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] nodes;
+
+    public MyHashMap() {
+        this.capacity = DEFAULT_START_CAPACITY;
+        this.threshold = (int) (DEFAULT_LOAD_FACTOR * capacity);
+        this.nodes = new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resizeList();
+        }
+        Node<K, V> newNode = new Node<>(key, value);
+        int index = getIndex(key);
+        Node<K, V> lastNode = getLastCollisionNode(index);
+        if (lastNode == null) {
+            nodes[index] = newNode;
+            size++;
+        } else {
+            putNewNode(newNode, nodes[index]);
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int index = getIndex(key);
+        Node<K, V> firsNodeOnIndex = nodes[index];
+        if (firsNodeOnIndex == null) {
+            return null;
+        }
+        if (firsNodeOnIndex.next == null) {
+            return firsNodeOnIndex.value;
+        }
```

Comment:
This is not correct. We must compare keys.

Reply:
++, removed this extra condition